core/src/bufr/BufrReader/Exports/Variables/SensorScanAngleVariable.cpp

```diff
@@ -16,13 +17,80 @@ namespace
     namespace ConfKeys
     {
         const char* FieldOfViewNumber = "fieldOfViewNumber";
+        const char* FieldOfRegardNumber = "fieldOfRegardNumber"; // optional
         const char* ScanStart = "scanStart";
         const char* ScanStep = "scanStep";
-        const char* ScanStepAdjust = "scanStepAdjust";
+        const char* ScanStepAdjust = "scanStepAdjust"; // optinal 
         const char* Sensor = "sensor";
     }  // namespace ConfKeys
 
     const std::vector<std::string> FieldNames = {ConfKeys::FieldOfViewNumber };
+
+    static const std::array<float, 9> fov_dist = {
+        2.71510e-2f, 1.91986e-2f, 2.71510e-2f,
+        1.91986e-2f, 0.0f,        1.91986e-2f,
+        2.71510e-2f, 1.91986e-2f, 2.71510e-2f
+    }; // unit is radians
+    static const std::array<float, 9> fov_ang = {
+        4.77057f, 3.98517f, 3.19977f,
+        5.55597f, 0.0f,     2.41437f,
+        0.05818f, 0.84358f, 1.62897f
+    }; // unit is radians
+
+    // Degrees to radians conversion
+    constexpr float degToRad(float degrees) {
+        return degrees * (M_PI / 180.0f);
+    }
+
+    // Radians to degree conversion
+    constexpr float radToDeg(float radians) {
+        return radians * (180.0f / M_PI);
+    }
+
+    // Compute scan angle for IASI 
+    void computeIasiScanAngles(std::vector<float>& scanang, const std::vector<int>& fovn,
+                               const std::vector<int>& scanpos, float start, float step, float stepAdj)
+    {
+        float tmp = -stepAdj;
+        for (size_t idx = 0; idx < fovn.size(); ++idx)
+       	{
+            if (scanpos[idx] % 2 == 1) tmp = stepAdj;
+            scanang[idx] = start + static_cast<float>((fovn[idx] - 1) / 4) * step + tmp;
+        }
```

Comment:
Computation of **tmp** is not correct. It will only be -stepAdj until it enters the **if** statement the very first time. It will be positive forever afterword.

Reply:
True!!  Fixed.